diff -up ./CMakeLists.txt.orig ./CMakeLists.txt
--- ./CMakeLists.txt.orig	2022-09-16 15:31:03.000000000 +0200
+++ ./CMakeLists.txt	2023-01-31 12:07:58.392962248 +0100
@@ -4,10 +4,14 @@ project(pvr.vbox)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR})
 
 find_package(Kodi REQUIRED)
+find_package(TinyXML2 REQUIRED)
 
 include_directories(${KODI_INCLUDE_DIR}/.. # Hack way with "/..", need bigger Kodi cmake rework to match right include ways
+                    ${TINYXML2_INCLUDE_DIRS}
                     ${PROJECT_SOURCE_DIR})
 
+set(DEPLIBS ${TINYXML2_LIBRARIES})
+
 # Sources and headers
 set(VBOX_SOURCES
                 src/addon.h
@@ -79,13 +83,6 @@ source_group("Source Files\\vbox\\respon
 source_group("Source Files\\timeshift" FILES ${VBOX_SOURCES_TIMESHIFT})
 source_group("Source Files\\xmltv" FILES ${VBOX_SOURCES_XMLTV})
 
-# Library files
-set(VBOX_LIBRARIES_TINYXML2
-                lib/tinyxml2/tinyxml2.h
-                lib/tinyxml2/tinyxml2.cpp)
-
-source_group("Library Files\\tinyxml2" FILES ${VBOX_LIBRARIES_TINYXML2})
-
 # Resource files
 set(VBOX_RESOURCES
                 README.md
@@ -105,8 +102,7 @@ list(APPEND VBOX_SOURCES
                 ${VBOX_SOURCES_VBOX_RESPONSE}
                 ${VBOX_SOURCES_TIMESHIFT}
                 ${VBOX_SOURCES_XMLTV}
-                ${VBOX_RESOURCES}
-                ${VBOX_LIBRARIES_TINYXML2})
+                ${VBOX_RESOURCES})
 
 build_addon(pvr.vbox VBOX DEPLIBS)
 
diff -up ./debian/control.orig ./debian/control
--- ./debian/control.orig	2022-09-16 15:31:03.000000000 +0200
+++ ./debian/control	2023-01-31 12:07:58.392962248 +0100
@@ -1,7 +1,7 @@
 Source: kodi-pvr-vbox
 Priority: extra
 Maintainer: Sam Stenvall <sam.stenvall@kodi.tv>
-Build-Depends: debhelper (>= 9.0.0), cmake,
+Build-Depends: debhelper (>= 9.0.0), cmake, libtinyxml2-dev,
                kodi-addon-dev
 Standards-Version: 4.1.2
 Section: libs
diff -up ./depends/common/tinyxml2/flags.txt.orig ./depends/common/tinyxml2/flags.txt
--- ./depends/common/tinyxml2/flags.txt.orig	2023-01-31 12:07:51.713867945 +0100
+++ ./depends/common/tinyxml2/flags.txt	2023-01-31 12:07:58.392962248 +0100
@@ -0,0 +1 @@
+-Dtinyxml2_BUILD_TESTING:BOOL=OFF
diff -up ./depends/common/tinyxml2/tinyxml2.sha256.orig ./depends/common/tinyxml2/tinyxml2.sha256
--- ./depends/common/tinyxml2/tinyxml2.sha256.orig	2023-01-31 12:07:51.714867959 +0100
+++ ./depends/common/tinyxml2/tinyxml2.sha256	2023-01-31 12:07:58.392962248 +0100
@@ -0,0 +1 @@
+cc2f1417c308b1f6acc54f88eb70771a0bf65f76282ce5c40e54cfe52952702c
diff -up ./depends/common/tinyxml2/tinyxml2.txt.orig ./depends/common/tinyxml2/tinyxml2.txt
--- ./depends/common/tinyxml2/tinyxml2.txt.orig	2023-01-31 12:07:51.714867959 +0100
+++ ./depends/common/tinyxml2/tinyxml2.txt	2023-01-31 12:07:58.393962262 +0100
@@ -0,0 +1 @@
+tinyxml2 https://github.com/leethomason/tinyxml2/archive/refs/tags/9.0.0.tar.gz
diff -up ./FindTinyXML2.cmake.orig ./FindTinyXML2.cmake
--- ./FindTinyXML2.cmake.orig	2023-01-31 12:07:35.484638804 +0100
+++ ./FindTinyXML2.cmake	2023-01-31 12:07:58.392962248 +0100
@@ -0,0 +1,27 @@
+# - Find TinyXML2
+# Find the native TinyXML2 includes and library
+#
+#   TINYXML2_FOUND        - True if TinyXML2 found.
+#   TINYXML2_INCLUDE_DIRS - where to find tinyxml2.h, etc.
+#   TINYXML2_LIBRARIES    - List of libraries when using TinyXML2.
+#
+
+find_package(PkgConfig)
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(PC_TINYXML2 tinyxml2 QUIET)
+endif()
+
+find_path(TINYXML2_INCLUDE_DIR tinyxml2.h PATHS ${PC_TINYXML2_INCLUDEDIR})
+find_library(TINYXML2_LIBRARY NAMES tinyxml2 PATHS ${PC_TINYXML2_LIBDIR})
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(TinyXML2
+                                  REQUIRED_VARS TINYXML2_LIBRARY TINYXML2_INCLUDE_DIR
+                                  VERSION_VAR ${PC_TINYXML2_VERSION})
+
+if(TINYXML2_FOUND)
+  set(TINYXML2_INCLUDE_DIRS ${TINYXML2_INCLUDE_DIR})
+  set(TINYXML2_LIBRARIES ${TINYXML2_LIBRARY})
+endif()
+
+mark_as_advanced(TINYXML2_INCLUDE_DIR TINYXML2_LIBRARY)
diff -up ./src/vbox/CategoryGenreMapper.cpp.orig ./src/vbox/CategoryGenreMapper.cpp
--- ./src/vbox/CategoryGenreMapper.cpp.orig	2022-09-16 15:31:03.000000000 +0200
+++ ./src/vbox/CategoryGenreMapper.cpp	2023-01-31 12:07:58.394962276 +0100
@@ -16,7 +16,7 @@
 #include <kodi/General.h>
 #include <kodi/addon-instance/pvr/EPG.h>
 #include <kodi/tools/StringUtils.h>
-#include <lib/tinyxml2/tinyxml2.h>
+#include <tinyxml2.h>
 
 // using namespace xmltv;
 using namespace kodi::tools;
@@ -68,7 +68,7 @@ bool CategoryGenreMapper::LoadCategoryTo
     std::unique_ptr<std::string> contents = utilities::ReadFileContents(fileHandle);
 
     // Try to parse the document
-    if (document.Parse(contents->c_str(), contents->size()) != XML_NO_ERROR)
+    if (document.Parse(contents->c_str(), contents->size()) != XML_SUCCESS)
       throw vbox::InvalidXMLException("XML parsing failed: " + std::string(document.ErrorName()));
 
     // Create mappings
diff -up ./src/vbox/GuideChannelMapper.cpp.orig ./src/vbox/GuideChannelMapper.cpp
--- ./src/vbox/GuideChannelMapper.cpp.orig	2022-09-16 15:31:03.000000000 +0200
+++ ./src/vbox/GuideChannelMapper.cpp	2023-01-31 12:07:58.394962276 +0100
@@ -13,7 +13,7 @@
 
 #include <memory>
 
-#include <lib/tinyxml2/tinyxml2.h>
+#include <tinyxml2.h>
 
 using namespace vbox;
 using namespace tinyxml2;
@@ -74,7 +74,7 @@ void GuideChannelMapper::Load()
     std::unique_ptr<std::string> contents = utilities::ReadFileContents(fileHandle);
 
     // Try to parse the document
-    if (document.Parse(contents->c_str(), contents->size()) != XML_NO_ERROR)
+    if (document.Parse(contents->c_str(), contents->size()) != XML_SUCCESS)
       throw vbox::InvalidXMLException("XML parsing failed: " + std::string(document.ErrorName()));
 
     // Create mappings
diff -up ./src/vbox/response/Content.cpp.orig ./src/vbox/response/Content.cpp
--- ./src/vbox/response/Content.cpp.orig	2022-09-16 15:31:03.000000000 +0200
+++ ./src/vbox/response/Content.cpp	2023-01-31 12:07:58.394962276 +0100
@@ -13,7 +13,7 @@
 #include "../../xmltv/Utilities.h"
 #include "../Channel.h"
 
-#include <lib/tinyxml2/tinyxml2.h>
+#include <tinyxml2.h>
 
 using namespace tinyxml2;
 using namespace vbox;
diff -up ./src/vbox/response/Response.cpp.orig ./src/vbox/response/Response.cpp
--- ./src/vbox/response/Response.cpp.orig	2022-09-16 15:31:03.000000000 +0200
+++ ./src/vbox/response/Response.cpp	2023-01-31 12:07:58.394962276 +0100
@@ -31,7 +31,7 @@ Response::~Response()
 void Response::ParseRawResponse(const std::string& rawResponse)
 {
   // Try to parse the response as XML
-  if (m_document->Parse(rawResponse.c_str(), rawResponse.size()) != XML_NO_ERROR)
+  if (m_document->Parse(rawResponse.c_str(), rawResponse.size()) != XML_SUCCESS)
     throw vbox::InvalidXMLException("XML parsing failed: " + std::string(m_document->ErrorName()));
 
   // Parse the response status
diff -up ./src/vbox/response/Response.h.orig ./src/vbox/response/Response.h
--- ./src/vbox/response/Response.h.orig	2022-09-16 15:31:03.000000000 +0200
+++ ./src/vbox/response/Response.h	2023-01-31 12:07:58.394962276 +0100
@@ -11,7 +11,7 @@
 #include <memory>
 #include <string>
 
-#include <lib/tinyxml2/tinyxml2.h>
+#include <tinyxml2.h>
 
 namespace vbox
 {
diff -up ./src/xmltv/Guide.cpp.orig ./src/xmltv/Guide.cpp
--- ./src/xmltv/Guide.cpp.orig	2022-09-16 15:31:03.000000000 +0200
+++ ./src/xmltv/Guide.cpp	2023-01-31 12:07:58.394962276 +0100
@@ -15,7 +15,7 @@
 #include <algorithm>
 
 #include <kodi/tools/StringUtils.h>
-#include <lib/tinyxml2/tinyxml2.h>
+#include <tinyxml2.h>
 
 using namespace kodi::tools;
 using namespace xmltv;
diff -up ./src/xmltv/Programme.cpp.orig ./src/xmltv/Programme.cpp
--- ./src/xmltv/Programme.cpp.orig	2022-09-16 15:31:03.000000000 +0200
+++ ./src/xmltv/Programme.cpp	2023-01-31 12:07:58.394962276 +0100
@@ -10,7 +10,7 @@
 
 #include "Utilities.h"
 
-#include <lib/tinyxml2/tinyxml2.h>
+#include <tinyxml2.h>
 
 using namespace xmltv;
 using namespace tinyxml2;
diff -up ./src/xmltv/Utilities.cpp.orig ./src/xmltv/Utilities.cpp
--- ./src/xmltv/Utilities.cpp.orig	2022-09-16 15:31:03.000000000 +0200
+++ ./src/xmltv/Utilities.cpp	2023-01-31 12:07:58.394962276 +0100
@@ -14,7 +14,7 @@
 #include <sstream>
 #include <stdexcept>
 
-#include <lib/tinyxml2/tinyxml2.h>
+#include <tinyxml2.h>
 
 using namespace xmltv;
 
